File: 2.2/C#/Numbers.cs
namespace Lab
{
    public class Numbers : Line
    {
        public Numbers(string input) : base(input) { }

        public override int GetLength()
        {
            return value.Length;
        }

        public override void Replace()
        {
            value = value.Replace("3", "11");
        }
    }
}

File: 2.2/C++/Numbers.cpp
#include "Numbers.h"

Numbers::Numbers(const std::string& input) : Line(input) {}

size_t Numbers::GetLength() const {
    return value.length();
}

void Numbers::Replace() {
    size_t pos = 0;
    while ((pos = value.find("3", pos)) != std::string::npos) {
        value.replace(pos, 1, "11");
        pos += 2;
    }
}
